content/blog/2023-10-18-skywalking-toolkit-trace/index.md

```diff
@@ -0,0 +1,285 @@
+---
+title: "Detailed explanation of SkyWalking Go Toolkit Trace"
+date: 2023-10-18
+author: "Alipebt"
+description: "This article shows the introduction and usage of toolkit trace provided by Skywalking Go to users."
+
+---
+
+## Background
+
+SkyWalking Go is an open-source, non-intrusive Golang agent used for monitoring, tracing, and data collection within distributed systems. It enables users to observe the flow and latency of requests within the system, collect performance data from various system components for performance monitoring, and troubleshoot issues by tracing the complete path of requests.
+
+In version v0.3.0, Skywalking Go introduced the toolkit trace tool. Trace APIs allow users to include critical operations, functions, or services in the tracing scope in situations where plugins do not support them. This inclusion enables tracking and monitoring of these operations and can be used for fault analysis, diagnosis, and performance monitoring.
+
+Before diving into this, you can learn how to use the Skywalking Go agent by referring to the [SkyWalking Go Agent Quick Start Guide](https://skywalking.apache.org/zh/2023-06-01-quick-start-with-skywalking-go-agent/).
+
+The following sections will explain how to use these interfaces in specific scenarios.
+
+## Introducing the Trace Toolkit
+
+Execute the following command in the project's root directory:
+
+```shell
+go get github.com/apache/skywalking-go/toolkit
+```
+
+To use the toolkit trace interface, you need to import the package into your project:
+
+```go
+"github.com/apache/skywalking-go/toolkit/trace"
+```
+
+## Manual Tracing
+
+A Span is the fundamental unit of an operation in Tracing. It represents an operation within a specific timeframe, such as a request, a function call, or a specific action. It records essential information about a particular operation, including start and end times, the operation's name, tags (key-value pairs), and relationships between operations. Multiple Spans can form a hierarchical structure.
+
+In situations where Skywalking-go doesn't support a particular framework, users can manually create Spans to obtain tracing information.
+
+(Here, I have removed the originally supported frameworks for the sake of the example. If needed, please replace them with unsupported frameworks.)
```

Comment:
```suggestion
```suggestion
(Here, I have removed the supported frameworks for the sake of the example. These are only examples. You should reference this when using the APIs in private and/or unsupported frameworks)
```
```